src/profile/v100/entity/display_name.rs

```diff
@@ -39,14 +48,10 @@ impl DisplayName {
         if value.is_empty() {
             return Err(CommonError::InvalidDisplayNameEmpty);
         }
-        if value.len() > Self::MAX_LEN {
-            return Err(CommonError::InvalidDisplayNameTooLong {
-                expected: Self::MAX_LEN as u64,
-                found: value.len() as u64,
-            });
-        }
 
-        Ok(Self { value })
+        Ok(Self {
+            value: value.chars().take(Self::MAX_LEN).collect(),
```

Comment:
Since we do this in two places, might as well write a func or method for it, put it in `string_utils` and add tests for it. 

You can perhaps extend the trait `StrExt`